lib/services/suppressions-service.js

```diff
@@ -0,0 +1,282 @@
+/**
+ * @fileoverview Manages the suppressed violations.
+ * @author Iacovos Constantinou
+ */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+const fs = require("node:fs");
+const path = require("node:path");
+const { calculateStatsPerFile } = require("../eslint/eslint-helpers");
+
+//-----------------------------------------------------------------------------
+// Exports
+//-----------------------------------------------------------------------------
+
+/**
+ * Manages the suppressed violations.
+ */
+class SuppressionsService {
+    filePath = "";
+    cwd = "";
+
+    /**
+     * Creates a new instance of SuppressionsService.
+     * @param {Object} options The options.
+     * @param {string} [options.filePath] The location of the suppressions file.
+     * @param {string} [options.cwd] The current working directory.
+     */
+    constructor({ filePath, cwd }) {
+        this.filePath = filePath;
+        this.cwd = cwd;
+    }
+
+    /**
+     * Updates the suppressions file based on the provided lint results.
+     * It goes through each file and counts the number of violations per rule.
+     * Warnings are ignored.
+     * @param {LintResult[]} results The lint results.
+     * @returns {Promise<void>}
+     */
+    suppressAll(results) {
+        const output = {};
+
+        for (const result of results) {
+            const relativeFilePath = this.getRelativeFilePath(result.filePath);
+            const violationsByRule = SuppressionsService.countViolationsByRule(result.messages);
+
+            output[relativeFilePath] = violationsByRule;
+        }
+
+        return this.save(output);
+    }
```

Comment:
Similar to https://github.com/eslint/eslint/pull/19159#discussion_r1927534339. This overwrites the suppressions file. I think it should only create/update entries for the files that were linted in the current run and leave the others unchanged.